packages/editor-ui/src/stores/ndv.store.ts

```diff
@@ -13,316 +12,440 @@ import {
+	const setDraggableTarget = (target: NDVState['draggable']['activeTarget']): void => {
+		draggable.value.activeTarget = target;
+	};
 
-			return null;
-		},
-		expressionTargetItem(): TargetItem | null {
-			if (this.getHoveringItem) {
-				return this.getHoveringItem;
-			}
+	const setMappingTelemetry = (telemetry: { [key: string]: string | number | boolean }): void => {
+		mappingTelemetry.value = { ...mappingTelemetry.value, ...telemetry };
+	};
 
-			if (this.expressionOutputItemIndex && this.ndvInputNodeName) {
-				return {
-					nodeName: this.ndvInputNodeName,
-					runIndex: this.ndvInputRunIndex ?? 0,
-					outputIndex: this.ndvInputBranchIndex ?? 0,
-					itemIndex: this.expressionOutputItemIndex,
-				};
-			}
+	const resetMappingTelemetry = (): void => {
+		mappingTelemetry.value = {};
+	};
 
-			return null;
-		},
-		isNDVOpen(): boolean {
-			return this.activeNodeName !== null;
-		},
-		ndvNodeInputNumber() {
-			const returnData: { [nodeName: string]: number[] } = {};
-			const workflow = useWorkflowsStore().getCurrentWorkflow();
-			const activeNodeConections = (
-				workflow.connectionsByDestinationNode[this.activeNode?.name || ''] ?? {}
-			).main;
-
-			if (!activeNodeConections || activeNodeConections.length < 2) return returnData;
-
-			for (const [index, connection] of activeNodeConections.entries()) {
-				for (const node of connection) {
-					if (!returnData[node.node]) {
-						returnData[node.node] = [];
-					}
-					returnData[node.node].push(index + 1);
-				}
-			}
+	const setHoveringItem = (item: TargetItem | null): void => {
+		if (item) setTableHoverOnboarded();
+		hoveringItem.value = item;
+	};
 
-			return returnData;
-		},
-	},
-	actions: {
-		setActiveNodeName(nodeName: string | null): void {
-			this.activeNodeName = nodeName;
-		},
-		setInputNodeName(nodeName: string | undefined): void {
-			this.input = {
-				...this.input,
-				nodeName,
-			};
-		},
-		setInputRunIndex(run?: number): void {
-			this.input = {
-				...this.input,
-				run,
-			};
-		},
-		setMainPanelDimensions(params: {
-			panelType: string;
-			dimensions: { relativeLeft?: number; relativeRight?: number; relativeWidth?: number };
-		}): void {
-			this.mainPanelDimensions = {
-				...this.mainPanelDimensions,
-				[params.panelType]: {
-					...this.mainPanelDimensions[params.panelType],
-					...params.dimensions,
+	const setNDVBranchIndex = (e: { pane: NodePanelType; branchIndex: number }): void => {
+		if (e.pane === 'input') {
+			input.value.branch = e.branchIndex;
+		} else {
+			output.value.branch = e.branchIndex;
+		}
+	};
+
+	const setNDVPanelDataIsEmpty = (params: {
+		panel: NodePanelType;
+		isEmpty: boolean;
+	}): void => {
+		if (params.panel === 'input') {
+			input.value.data.isEmpty = params.isEmpty;
+		} else {
+			output.value.data.isEmpty = params.isEmpty;
+		}
+	};
+
+	const setMappingOnboarded = () => {
+		isMappingOnboarded.value = true;
+		useStorage(LOCAL_STORAGE_MAPPING_IS_ONBOARDED).value = 'true';
```

Comment:
We should keep the `useStorage` values outside of the function, as variables, to avoid registering a listener every time we call `setMappingOnboarded`.